### ISSUE TITLE
how to implement optgroup ?

### ISSUE THREAD

**Issue by pratikt-cuelogic** [benbowes/react-responsive-select#93]
how to implement optgroup ?
reference: https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_optgroup

**Comment by benbowes** [comment]
It is not currently possible

**Comment by benbowes** [comment]
If the is interest in this, I will implement it for single select

**Comment by pratikt-cuelogic** [comment]
I would really appreciate if we could have it

**Comment by benbowes** [comment]
Will do this

**Comment by benbowes** [comment]
Hi @pratikt-cuelogic,

I have implemented this as an option header, not an optGroup. This way I can handle both option headings utilising position: sticky, whilst not degrading screen-reader output or keyboard access.

e.g. using `optHeader: true`
```
options={[
  { text: 'Cheap', optHeader: true },
  {
    value: 'alfa-romeo',
    text: 'Alfa Romeo',
  },
  {
    value: 'fiat',
    text: 'Fiat',
  },
  {
    value: 'subaru',
    text: 'Subaru',
  },
  {
    value: 'suzuki',
    text: 'Suzuki',
  },
  { text: 'Expensive', optHeader: true },
  {
    value: 'bmw',
    text: 'BMW',
  },
  {
    value: 'ferrari',
    text: 'Ferrari',
  },
  {
    value: 'mercedes',
    text: 'Mercedes',
  },
  {
    value: 'tesla',
    text: 'Tesla',
  },
  {
    value: 'volvo',
    text: 'Volvo',
  },
  {
    value: 'zonda',
    text: 'Zonda',
  },
]}
```

Available in react-responsive-select@4.5.3
